server/data/mappers/createInductionMapper.ts

```diff
@@ -0,0 +1,43 @@
+import type { CreateInductionRequest } from 'educationAndWorkPlanApiClient'
+import type { CreateInductionDto } from 'dto'
+
+const toCreateInductionRequest = (createInductionDto: CreateInductionDto): CreateInductionRequest => {
+  return {
+    ...createInductionDto,
+    workOnRelease: {
+      ...createInductionDto.workOnRelease,
+    },
+    previousQualifications: createInductionDto.previousQualifications
+      ? {
+          ...createInductionDto.previousQualifications,
+        }
+      : undefined,
+    previousTraining: createInductionDto.previousTraining
+      ? {
+          ...createInductionDto.previousTraining,
+        }
+      : undefined,
+    previousWorkExperiences: createInductionDto.previousWorkExperiences
+      ? {
+          ...createInductionDto.previousWorkExperiences,
+        }
+      : undefined,
+    inPrisonInterests: createInductionDto.inPrisonInterests
+      ? {
+          ...createInductionDto.inPrisonInterests,
+        }
+      : undefined,
+    personalSkillsAndInterests: createInductionDto.personalSkillsAndInterests
+      ? {
+          ...createInductionDto.personalSkillsAndInterests,
+        }
+      : undefined,
+    futureWorkInterests: createInductionDto.futureWorkInterests
+      ? {
+          ...createInductionDto.futureWorkInterests,
+        }
+      : undefined,
+  }
+}
```

Comment:
Personally I quite like the "long hand" nature of this because it is clear and explicit about the cardinality of the fields 👍 

However you don't need to be as explicit about it. This would also work:
```
const toCreateInductionRequest = (createInductionDto: CreateInductionDto): CreateInductionRequest => {
  return { ...createInductionDto }
}
```
(As above, personally I think it's too concise and masks some of the detail re: cardinality)

The reason I had to do this approach in my mapper is because for each of the child types, as well as copying it's fields (ie. the `...` notation), I also had to cast/convert the audit date fields from `string`s to `Date`s

Up to you if you want to change it to the (very) concise version I mentioned above (but for me this longer version is better for readability 😁 ) - I just thought I'd explain why my mapper is written the way it is 👍

Reply:
Ah that's interesting. I quite like the concise version, as it's less error prone. The cardinality can be easily worked out by looking at the objects involved.